crates/supervisor/types/src/types.rs

```diff
@@ -134,3 +136,33 @@ pub struct ManagedEvent {
     /// Indicates an update to the derivation origin
     pub derivation_origin_update: Option<BlockInfo>,
 }
+
+impl fmt::Display for ManagedEvent {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut parts = Vec::new();
+        if let Some(ref reset) = self.reset {
+            parts.push(format!("reset: {reset}"));
+        }
+        if let Some(ref block) = self.unsafe_block {
+            parts.push(format!("unsafe_block: {block}"));
+        }
+        if let Some(ref pair) = self.derivation_update {
+            parts.push(format!("derivation_update: {pair}"));
+        }
+        if let Some(ref pair) = self.exhaust_l1 {
+            parts.push(format!("exhaust_l1: {pair}"));
+        }
+        if let Some(ref replacement) = self.replace_block {
+            parts.push(format!("replace_block: {replacement}"));
+        }
+        if let Some(ref origin) = self.derivation_origin_update {
+            parts.push(format!("derivation_origin_update: {origin}"));
+        }
+
+        if parts.is_empty() {
+            write!(f, "ManagedEvent {{ none }}")
+        } else {
+            write!(f, "ManagedEvent {{ {} }}", parts.join(", "))
+        }
+    }
+}
```

Comment:
this impl is redundant because it's basically the same as the debug impl. display impl usually doesn't show the type name. so rather just a comma separated list of the values without and object delimiters or such.
> 
```
// Similarly, implement `Display` for `Point2D`.
impl fmt::Display for Point2D {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        // Customize so only `x` and `y` are denoted.
        write!(f, "x: {}, y: {}", self.x, self.y)
    }
}
```

see how it's done in these docs https://doc.rust-lang.org/rust-by-example/hello/print/print_display.html

Reply:
Thanks for pointing that out. We can remove the outer ManagedEvent {{ ... }} wrapper and just display the fields directly, separated by commas.

I think Display implementation for `ManagedEvent` offers a practical advantage. It omits fields that are `None`, showing only the fields that are actually present in the event. This makes logs and output much more concise and focused, especially since `ManagedEvent` is a sparse struct where most fields are often None. 

Let me know what you think.

Reply:
yes please leave out the type name and curly brackets

Reply:
Thanks. I have pushed the changes